phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -527,6 +535,28 @@ public static boolean isMaxLookbackTimeEnabled(long maxLookbackTime){
         return maxLookbackTime > 0L;
     }
 
+    public static long getMaxLookbackAge(ObserverContext<RegionCoprocessorEnvironment> c) {
+        TableName tableName = c.getEnvironment().getRegion().getRegionInfo().getTable();
+        String fullTableName = tableName.getNameAsString();
+        Configuration conf = c.getEnvironment().getConfiguration();
+        PTable table;
+        try(PhoenixConnection conn = QueryUtil.getConnectionOnServer(
+                conf).unwrap(PhoenixConnection.class)) {
+            table = conn.getTableNoCache(fullTableName);
+        }
+        catch (Exception e) {
+            if (e instanceof TableNotFoundException) {
+                LOGGER.debug("Ignoring HBase table that is not a Phoenix table: "
+                        + fullTableName);
+                // non-Phoenix HBase tables won't be found, do nothing
```

Comment:
We need to see if the HBase table name for a given Phoenix table has changed by the online data format change feature. We need query the system catalog using WHERE  PHYSICAL_TABLE_NAME = '<this HBase table name>' to find the Phoenix table name.

Reply:
Thanks I missed this scenario. Wondering if same can happen in here also: https://github.com/apache/phoenix/blob/master/phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java#L601-L614.

Reply:
If we have a table `T1` and we do a transform on it, seems like SYSCAT will have following two records: 

- A record with `TABLE_NAME` as `T1_1` and `PHYSICAL_TABLE_NAME` also as `T1_1`.
- A record with `TABLE_NAME` as `T1` and `PHYSICAL_TABLE_NAME` as `T1_1`.

As we are looking up Phoenix table from HBase table shall we skip the records where `TABLE_NAME` == `PHYSICAL_TABLE_NAME` and `PHYSICAL_TABLE_NAME is NOT NULL`?

Reply:
As above WHERE clause was causing FULL SCAN over SYSCAT and to avoid that went with an alternate approach. Please take a look. Thanks

Reply:
> What is the alternative approach?

In `TransformClient` we derive new table name from existing table name. I saw we are appending sequence num to eixtsing table's full name separated by underscore. I added a logic to derive logical table name from physical table name by checking if physical table is underscore separated and if part of table name before underscore is valid logical table name in SYSCAT along with checking if the logical table found has physical table as original physical table name.

I thought this will avoid FULL SCAN along max be one RPC call but even any READ operation on any table even if we create an index will be at least one RPC call. Please correct me if I misunderstood something. Thanks

Reply:
Based on a side conversation with @sanjeet006py, I learned that  the IT tests for Transform is not running currently due to missing @Category annotation and some of these tests are broken. We need to fix the transform tool issues in a separate Jira. We can assume that the transform tool is not supported yet and the name of a Phoenix table and the name of its HBase table are the same for now. When we fix transform tool issues, we need to handle the table name mismatch issues in general.

Reply:
> I saw we are appending sequence num to eixtsing table's full name separated by underscore
I see. We need to revisit this later to see if it is safe to rely on this.